### PR TITLE
Sign Up Feature implemented

### PR DESCRIPTION
**Pull request by irchik**
need code review

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,4 +3,5 @@ source 'https://rubygems.org'
 gem 'howitzer' #, github: "strongqa/howitzer", branch: "develop"
 gem 'minitest-reporters', '>= 0.5.0' # is necessary in RubyMine with minitest (4.7.5)
 gem 'debase'         # are necessary for debugging
-gem 'ruby-debug-ide' # in RubyMine
\ No newline at end of file
+gem 'ruby-debug-ide' # in RubyMine
+gem 'selenium-webdriver'
```
Comment (romikoops) [inline Gemfile]: Selenium Webdriver included to howitzer. Why did you add it here?

File: config/default.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@
  maximized_window: true
 
  # Specify one of the following drivers: selenium, selenium_dev, webkit, poltergeist, sauce, testingbot, phantomjs
- driver: phantomjs
+ driver: selenium
```
Comment (romikoops) [inline config/default.yml]: please use phantomjs as default driver

File: features/functionality/auth/signup.feature [hunk 3]
```diff
@@ -116,17 +108,17 @@ Feature: Sign Up
...
+     """
 
   @p1
   Scenario: user can not sign up with duplicated email
-    Given registered user with data:
+    Given registered user with parameters:
       | user_name             | UNIQ_USER[:full_name] |
       | email                 | UNIQ_USER[:email]     |
       | password              | UNIQ_USER[:password]  |
+      |password_confirmation  | UNIQ_USER[:password]  |
```
Comment (romikoops) [inline features/functionality/auth/signup.feature]: password_confirmation is not required. Also, because 'data' is used in other tests, and we have other pull requests, I would recommend do not use 'parameters'

File: features/step_definitions/common_steps.rb [hunk 4]
```diff
@@ -33,7 +36,7 @@
   page.given.choose_menu(text)
 end
 
-When /^I fill form data on (.+) page:$/ do |page, table|
+When /^I fill form on (.+) page with data:$/ do |page, table|
```
Comment (romikoops) [inline features/step_definitions/common_steps.rb]: Have you changed all scenario with new text?

File: features/step_definitions/sign_up_steps.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+When /^I submit Sign up form on Sign up page$/ do
```
Comment (romikoops) [inline features/step_definitions/sign_up_steps.rb]: Use comments for sections, like https://github.com/irchik/howitzer_example/blob/master/features/step_definitions/common_steps.rb#L2

File: pages/home_page.rb [hunk 6]
```diff
@@ -6,6 +6,8 @@ class HomePage < WebPage
   validates :url, pattern: /\A(?:.*?:\/\/)?[^\/]*\/?\z/
 
   add_locator :article_link, lambda{|title| {xpath: ".//a[.='#{title}']"}}
+  add_locator :message,        css: "#flash_notice"
```
Comment (romikoops) [inline pages/home_page.rb]: it should be somewhere in menu for instance, because it is using on all pages

File: pages/sign_up_page.rb [hunk 7]
```diff
@@ -14,12 +13,11 @@ class SignUpPage < WebPage
 
   include MainMenu
 
-  def fill_form(user_name: nil, email: nil, password: nil, password_confirmation: nil)
```
Comment (romikoops) [inline pages/sign_up_page.rb]: Why  did you change old implementation?

File: features/functionality/auth/signup.feature (hunk 3 above)
Comment (irchik) [inline features/functionality/auth/signup.feature]: Should I change 
 Given registered user with parameters: -> + Given registered user with data :

File: Gemfile (hunk 1 above)
Comment (irchik) [inline Gemfile]: fixed

File: config/default.yml (hunk 2 above)
Comment (irchik) [inline config/default.yml]: fixed

File: features/functionality/auth/signup.feature (hunk 3 above)
Comment (irchik) [inline features/functionality/auth/signup.feature]: password_confirmaion is a required field. test can`t pass without it.'parameters' was changed to `data`

File: features/step_definitions/common_steps.rb (hunk 4 above)
Comment (irchik) [inline features/step_definitions/common_steps.rb]: fixed. all scenarios were changed

File: pages/home_page.rb (hunk 6 above)
Comment (irchik) [inline pages/home_page.rb]: fixed

File: Gemfile [hunk 8]
```diff
@@ -2,8 +2,6 @@ source 'https://rubygems.org'
 
 gem 'nokogiri'
 gem 'howitzer' #, github: "strongqa/howitzer", branch: "develop"
-gem 'gherkin', platforms: :ruby
 gem 'minitest-reporters', '>= 0.5.0' # is necessary in RubyMine with minitest (4.7.5)
-#gem 'debase'         # are necessary for debugging
-#gem 'ruby-debug-ide' # in RubyMine
-gem 'capybara-webkit'
\ No newline at end of file
+gem 'debase'         # are necessary for debugging
```
Comment (romikoops) [inline Gemfile]: comment it please

File: Gemfile [hunk 9]
```diff
@@ -2,8 +2,6 @@ source 'https://rubygems.org'
...
+gem 'ruby-debug-ide' # in RubyMine
```
Comment (romikoops) [inline Gemfile]: comment it please

File: config/default.yml [hunk 10]
```diff
@@ -22,7 +22,7 @@
  maximized_window: true
 
  # Specify one of the following drivers: selenium, selenium_dev, webkit, poltergeist, sauce, testingbot, phantomjs
- driver: phantomjs
+ driver: poltergeist
```
Comment (romikoops) [inline config/default.yml]: revert this change

File: features/functionality/auth/signup.feature [hunk 11]
```diff
@@ -18,12 +19,12 @@ Feature: Sign Up
...
   Scenario: user can sign up with correct credentials
     Given Sign up page of web application
-    When I fill form data on Sign up page:
+    When I fill Sign up form on Sign up page with data:
       | user_name             | UNIQ_USER[:full_name] |
       | email                 | UNIQ_USER[:email]     |
       | password              | UNIQ_USER[:password]  |
       | password_confirmation | UNIQ_USER[:password]  |
-    And I submit form on Sign up page
+    And I submit Sign up form on Sign up page
```
Comment (romikoops) [inline features/functionality/auth/signup.feature]: Why we have "Sign up form" here. What about   "When I fill form data on Sign up page:"

General rule - 1 form = form data, > 1 form = named form

File: features/functionality/auth/signup.feature [hunk 12]
```diff
@@ -115,27 +108,28 @@ Feature: Sign Up
...
 
-  @p1 @wip
-  Scenario: user can not sign up with duplicated email
+  @p1
+  Scenario: user cannot sign up with duplicated email
     Given registered user with data:
       | user_name             | UNIQ_USER[:full_name] |
       | email                 | UNIQ_USER[:email]     |
       | password              | UNIQ_USER[:password]  |
+      | password_confirmation | UNIQ_USER[:password]  |
```
Comment (romikoops) [inline features/functionality/auth/signup.feature]: password_confirmation here is extra, because it is enough password only to pass to 2 fields:

https://github.com/strongqa/howitzer_example/blob/master/features/functionality/auth/password_resetting.feature#L9

File: features/step_definitions/common_steps.rb [hunk 13]
```diff
@@ -10,26 +10,20 @@
...
-  step "I should receive confirmation instruction email for #{data[:email]} recipient"
-  step "I confirm #{data[:email]} account from confirmation instruction email"
-  step "I should see following text on Login page:","Your account was successfully confirmed."
-end
+  user = table.rows_hash.symbolize_keys
+  SignUpPage.open
+  SignUpPage.given.fill_form(table.rows_hash.symbolize_keys)
+  SignUpPage.given.submit_form
+  email = ConfirmationInstructionEmail.find_by_recipient(user[:email])
+  email.confirm_my_account
```
Comment (romikoops) [inline features/step_definitions/common_steps.rb]: Why you replaced Micha's code?

File: pages/home_page.rb [hunk 14]
```diff
@@ -6,6 +6,8 @@ class HomePage < WebPage
...
+  add_locator :log_oul_link,   css: "a[href='/users/sign_out']"
```
Comment (romikoops) [inline pages/home_page.rb]: spelling error log_ouT_link

File: pages/home_page.rb (hunk 6 above)
Comment (romikoops) [inline pages/home_page.rb]: Where did you fix it?

File: features/step_definitions/common_steps.rb [hunk 15]
```diff
@@ -83,9 +77,5 @@
 end
 
 Then /^I should be redirected to (.+) page$/ do |page|
-  page.given
-end
-
-Then /^I should see (.+) page$/ do |page|
```
Comment (romikoops) [inline features/step_definitions/common_steps.rb]: Why it was removed?

File: Gemfile (hunk 8 above)
Comment (irchik) [inline Gemfile]: fixed

File: Gemfile (hunk 9 above)
Comment (irchik) [inline Gemfile]: fixed

File: config/default.yml (hunk 10 above)
Comment (irchik) [inline config/default.yml]: fixed

File: features/functionality/auth/signup.feature (hunk 11 above)
Comment (irchik) [inline features/functionality/auth/signup.feature]: fixed

File: features/functionality/auth/signup.feature (hunk 12 above)
Comment (irchik) [inline features/functionality/auth/signup.feature]: fixed

File: features/step_definitions/common_steps.rb (hunk 13 above)
Comment (irchik) [inline features/step_definitions/common_steps.rb]: fixed .
